### DISCUSSION TITLE
Reduce JavaScript execution time

### DISCUSSION THREAD

**Discussion by carstenjaksch** [sveltejs/kit#11145] [Q&A]
Hi there,

I am currently building a new site and do some GTmetrix and PageSpeed Insights scans from time to time.

After every scan, PageSpeed Insights displays this file with the highest execution time of ~3.000 ms: `_app/immutable/chunks/scheduler.dc08184f.js`

I know that these results should not be considered the single source of truth. But I am eager ...

What does this file do? Why is it there and why does it take so long to execute?

**Comment by Mboy3000** [comment]
So, without actually getting into the nitty gritty of your source code it's hard to say for certain. However, a few things to keep in mind when it comes to the speed of execution for your programming is the files size and computational complexity of what it is trying to achieve. There is also the fact that JS is a scripting language using a interperter instead of a compiler so it is naturally going to be slower then lets say C.  and another thing to consider is are you running a website with multiple HTML and CSS and JS files or is it converted to be a single Page applicaiton. The SPA will be faster at the cost of SEO and the website will be slower but have a easier time with you SEO.  As far as I can tell, that seems to just be the name of a JS file you have. However, besides those general concepts it's difficult to say whats going on.  if you want to link me up to the repository, or just send me a screen shot of the file I can take a look at it for you.
